modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java

```diff
@@ -145,10 +145,10 @@ public Slot getSlot(String queueName, String nodeId) {
         String lockKey = queueName + SlotManagerClusterMode.class;
 
         synchronized (lockKey.intern()) {
-            slotToBeAssigned = getOverlappedSlot(nodeId, queueName);
+            slotToBeAssigned = getUnassignedSlot(queueName);
```

Comment:
Is it right to fetch an unassigned slot before checking overlapped ones ? the reasoning here was to ensure that overlapped slots are given priority when giving a slot. (So that before reaching an unassigned slot, overlaps can be handled)